### PR TITLE
Updating the K-NN Filters documentation due to recent enhancements in…

### PR DESCRIPTION
**Pull request by navneet1v**
### Description
Updating the K-NN Filters documentation due to recent enhancements in Efficient Filters

### Issues Resolved
Fixes https://github.com/opensearch-project/documentation-website/issues/4962

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _search-plugins/knn/filter-search-knn.md [hunk 1]
```diff
@@ -261,13 +261,16 @@ For more ways to construct a filter, see [Constructing a filter](#constructing-a
...
+Starting with k-NN plugin version 2.9, you can use `faiss` filters for k-NN searches for HNSW algorithm and with plugin version 2.10 you can do filtering with `IVF` algorithm too.
 
 When you specify a Faiss filter for a k-NN search, the Faiss algorithm decides whether to perform an exact k-NN search with pre-filtering or an approximate search with modified post-filtering. The algorithm uses the following variables:
 
 - N: The number of documents in the index.
 - P: The number of documents in the document subset after the filter is applied (P <= N).
 - k: The maximum number of vectors to return in the response.
+- R: number of results returned after doing the Filtered Approximate Nearest Neighbor Search
+- FilterThreshold: A per index level threshold defined via `knn.advanced.filtered_exact_search_threshold` setting( [Ref]({{site.url}}{{site.baseurl}}/search-plugins/knn/settings/)) to switch to exact search. (P <= FilterThreshold)
+- MaxDistanceComputations: The maximum number of distance computations allowed in exact search if FilterThreshold is not set. This value cannot be changed.
```
Comment (kolchfa-aws) [inline _search-plugins/knn/filter-search-knn.md:273]:
```suggestion
- max distance computations: The maximum number of distance computations allowed in exact search if f filtered threshold is not set. This value cannot be changed. 
```

File: _search-plugins/knn/settings.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ The k-NN plugin adds several new cluster settings.
 ## Cluster settings
 
 Setting | Default | Description
-:--- | :--- | :---
+:--- | :-- | :---
```
Comment (kolchfa-aws) [inline _search-plugins/knn/settings.md:15]:
```suggestion
:--- | :--- | :---
```

File: _search-plugins/knn/settings.md [hunk 3]
```diff
@@ -23,3 +23,4 @@ Setting | Default | Description
 `knn.plugin.enabled`| true | Enables or disables the k-NN plugin.
 `knn.model.index.number_of_shards`| 1 | Number of shards to use for the model system index, the OpenSearch index that stores the models used for Approximate k-NN Search.
 `knn.model.index.number_of_replicas`| 1 | Number of replica shards to use for the model system index. Generally, in a multi-node cluster, this should be at least 1 to increase stability.
+`knn.advanced.filtered_exact_search_threshold`| null | Threshold value for the filtered ids that will be used to switch to exact search during Filtered ANN Search. If number of filteredIds on a segment is less than the settings value, exact search will be done on the filtered Ids.
```
Comment (kolchfa-aws) [inline _search-plugins/knn/settings.md:26]:
```suggestion
`knn.advanced.filtered_exact_search_threshold`| null | Threshold value for the filtered IDs that is used to switch to exact search during filtered ANN search. If number of filtered IDs in a segment is less than this setting's value, exact search will be done on the filtered IDs. 
```

**Review by kolchfa-aws — COMMENTED** [review]
Thanks very much, @navneet1v! Some minor wording and variable name suggestions.

File: _search-plugins/knn/filter-search-knn.md [hunk 4]
```diff
@@ -261,16 +261,16 @@ For more ways to construct a filter, see [Constructing a filter](#constructing-a
...
 
 - N: The number of documents in the index.
 - P: The number of documents in the document subset after the filter is applied (P <= N).
 - k: The maximum number of vectors to return in the response.
-- R: number of results returned after doing the Filtered Approximate Nearest Neighbor Search
-- FilterThreshold: A per index level threshold defined via `knn.advanced.filtered_exact_search_threshold` setting( [Ref]({{site.url}}{{site.baseurl}}/search-plugins/knn/settings/)) to switch to exact search. (P <= FilterThreshold)
-- MaxDistanceComputations: The maximum number of distance computations allowed in exact search if FilterThreshold is not set. This value cannot be changed. 
+- R: The number of results returned after doing the Filtered Approximate Nearest Neighbor Search.
+- FT: An index-level threshold defined in the [`knn.advanced.filtered_exact_search_threshold` setting]({{site.url}}{{site.baseurl}}/search-plugins/knn/settings/) that specifies to switch to exact search.
```
Comment (kolchfa-aws) [inline _search-plugins/knn/filter-search-knn.md:272]:
```suggestion
- FT (filtered threshold): An index-level threshold defined in the [`knn.advanced.filtered_exact_search_threshold` setting]({{site.url}}{{site.baseurl}}/search-plugins/knn/settings/) that specifies to switch to exact search.
```

File: _search-plugins/knn/filter-search-knn.md [hunk 5]
```diff
@@ -261,16 +261,16 @@ For more ways to construct a filter, see [Constructing a filter](#constructing-a
...
+- MDC: The maximum number of distance computations allowed in exact search if `f` filtered threshold is not set. This value cannot be changed.
```
Comment (kolchfa-aws) [inline _search-plugins/knn/filter-search-knn.md:273]:
```suggestion
- MDC (max distance computations): The maximum number of distance computations allowed in exact search if `FT` (filtered threshold) is not set. This value cannot be changed.
```

**Review by kolchfa-aws — APPROVED** [review]
LGTM. Thanks for making the changes, @navneet1v! I'll arrange for an editorial review and then the PR will be ready to merge.

File: _search-plugins/knn/filter-search-knn.md [hunk 6]
```diff
@@ -13,7 +13,7 @@ To refine k-NN results, you can filter a k-NN search using one of the following
 
 - [Efficient k-NN filtering](#efficient-k-nn-filtering): This approach applies filtering _during_ the k-NN search, as opposed to before or after the k-NN search, which ensures that `k` results are returned (if there are at least `k` results in total). This approach is supported by the following engines:
   - Lucene engine with a Hierarchical Navigable Small World (HNSW) algorithm (k-NN plugin versions 2.4 and later) 
-  - Faiss engine with an HNSW algorithm (k-NN plugin versions 2.9 or later) 
+  - Faiss engine with an HNSW algorithm (k-NN plugin versions 2.9 or later) or IVF algorithm (k-NN plugin versions 2.10 or later)
```
Comment (natebower) [inline _search-plugins/knn/filter-search-knn.md:16]:
```suggestion
  - Faiss engine with an HNSW algorithm (k-NN plugin versions 2.9 and later) or IVF algorithm (k-NN plugin versions 2.10 and later)
```

Comment (natebower) [inline _search-plugins/knn/filter-search-knn.md:16]: Does IVF need to be defined?

File: _search-plugins/knn/filter-search-knn.md [hunk 7]
```diff
@@ -261,13 +261,16 @@ For more ways to construct a filter, see [Constructing a filter](#constructing-a
 
 ### Faiss k-NN filter implementation 
 
-Starting with k-NN plugin version 2.9, you can use `faiss` filters for k-NN searches.
+For k-NN searches, you can use `faiss` filters with an HNSW algorithm (k-NN plugin versions 2.9 or later) or IVF algorithm (k-NN plugin versions 2.10 or later).
```
Comment (natebower) [inline _search-plugins/knn/filter-search-knn.md:264]:
```suggestion
For k-NN searches, you can use `faiss` filters with an HNSW algorithm (k-NN plugin versions 2.9 and later) or IVF algorithm (k-NN plugin versions 2.10 and later).
```

File: _search-plugins/knn/filter-search-knn.md [hunk 8]
```diff
@@ -261,13 +261,16 @@ For more ways to construct a filter, see [Constructing a filter](#constructing-a
...
 
 When you specify a Faiss filter for a k-NN search, the Faiss algorithm decides whether to perform an exact k-NN search with pre-filtering or an approximate search with modified post-filtering. The algorithm uses the following variables:
 
 - N: The number of documents in the index.
 - P: The number of documents in the document subset after the filter is applied (P <= N).
 - k: The maximum number of vectors to return in the response.
+- R: The number of results returned after doing the Filtered Approximate Nearest Neighbor Search.
```
Comment (natebower) [inline _search-plugins/knn/filter-search-knn.md:271]:
```suggestion
- R: The number of results returned after performing the filtered approximate nearest neighbor search.
```

File: _search-plugins/knn/settings.md [hunk 9]
```diff
@@ -23,3 +23,4 @@ Setting | Default | Description
 `knn.plugin.enabled`| true | Enables or disables the k-NN plugin.
 `knn.model.index.number_of_shards`| 1 | Number of shards to use for the model system index, the OpenSearch index that stores the models used for Approximate k-NN Search.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:24]:
```suggestion
`knn.model.index.number_of_shards`| 1 | The number of shards to use for the model system index, the OpenSearch index that stores the models used for Approximate k-NN Search.
```

File: _search-plugins/knn/settings.md [hunk 10]
```diff
@@ -23,3 +23,4 @@ Setting | Default | Description
...
 `knn.model.index.number_of_replicas`| 1 | Number of replica shards to use for the model system index. Generally, in a multi-node cluster, this should be at least 1 to increase stability.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:25]:
```suggestion
`knn.model.index.number_of_replicas`| 1 | The number of replica shards to use for the model system index. Generally, in a multi-node cluster, this should be at least 1 to increase stability.
```

File: _search-plugins/knn/settings.md [hunk 11]
```diff
@@ -23,3 +23,4 @@ Setting | Default | Description
...
+`knn.advanced.filtered_exact_search_threshold`| null | Threshold value for the filtered IDs that is used to switch to exact search during filtered ANN search. If number of filtered IDs in a segment is less than this setting's value, exact search will be done on the filtered IDs.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:26]:
```suggestion
`knn.advanced.filtered_exact_search_threshold`| null | The threshold value for the filtered IDs that is used to switch to exact search during filtered ANN search. If the number of filtered IDs in a segment is less than this setting's value, exact search will be performed on the filtered IDs.  
```

Comment (natebower) [inline _search-plugins/knn/settings.md:26]: Here we use "ANN", above we use "Approximate k-NN Search", and earlier we spelled out "approximate nearest neighbor". Please ensure that we define ANN on first appearance and use only the abbreviation thereafter.

**Review by natebower — COMMENTED** [review]
@navneet1v @kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _search-plugins/knn/filter-search-knn.md (hunk 6 above)
Comment (navneet1v) [reply _search-plugins/knn/filter-search-knn.md:16]: Yes.

File: _search-plugins/knn/settings.md [hunk 12]
```diff
@@ -21,5 +21,6 @@ Setting | Default | Description
 `knn.memory.circuit_breaker.limit` | 50% | The native memory limit for native library indexes. At the default value, if a machine has 100 GB of memory and the JVM uses 32 GB, the k-NN plugin uses 50% of the remaining 68 GB (34 GB). If memory usage exceeds this value, k-NN removes the least recently used native library indexes.
 `knn.memory.circuit_breaker.enabled` | true | Whether to enable the k-NN memory circuit breaker.
 `knn.plugin.enabled`| true | Enables or disables the k-NN plugin.
-`knn.model.index.number_of_shards`| 1 | Number of shards to use for the model system index, the OpenSearch index that stores the models used for Approximate k-NN Search.
-`knn.model.index.number_of_replicas`| 1 | Number of replica shards to use for the model system index. Generally, in a multi-node cluster, this should be at least 1 to increase stability.
+`knn.model.index.number_of_shards`| 1 | The number of shards to use for the model system index, the OpenSearch index that stores the models used for Approximate Nearest Neighbor(ANN) Search.
```
Comment (kolchfa-aws) [inline _search-plugins/knn/settings.md:24]:
```suggestion
`knn.model.index.number_of_shards`| 1 | The number of shards to use for the model system index, the OpenSearch index that stores the models used for Approximate Nearest Neighbor (ANN) search.
```

**Pull request merged**